test/Label.test.js

```diff
@@ -418,6 +435,686 @@ describe('Network Label', function() {
+      },
+    });
+*/
+
+    done();
+  });
+
+
+  it('sets and uses font.multi in group options', function (done) {
+
+    /**
+     * Helper function for easily accessing font options in a node
+     */
+    var fontOption = (index) => {
+      var nodes = network.body.nodes;
+      return nodes[index].labelModule.fontOptions;
+    };
+
+
+    /**
+     * Helper function for easily accessing bold options in a node
+     */
+    var modBold = (index) => {
+      return fontOption(index).bold;
+    };
+
+
+    var dataNodes = [
+      {id: 1, label: '<b>1</b>', group: 'group1'},
+      {
+        // From example 1 in #3408
+        id: 6, 
+        label: '<i>\uf286</i> <b>\uf2cd</b> colored glyph icon',
+        shape: 'icon',
+        group: 'colored',
+        icon : { color: 'blue' },
+        font:
+        {
+          bold : { color : 'blue' },
+          ital : { color : 'green' }
+        }
+      },
+    ];
+  
+    // create a network
+    var container = document.getElementById('mynetwork');
+    var data = {
+        nodes: new vis.DataSet(dataNodes),
+        edges: []
+    };
+  
+    var options = {
+      groups: {
+        group1: {
+          font: {
+            multi: true,
+            color: 'red'
+          },
+        },
+        colored :
+        {
+          // From example 1 in 3408
+          icon :
+          {
+            face : 'FontAwesome',
+            code : '\uf2b5',
+          },
+          font:
+          {
+            face : 'FontAwesome',
+            multi: true,
+            bold : { mod : '' },
+            ital : { mod : '' }
+          }
+        },
+      },
+    };
+  
+    var network = new vis.Network(container, data, options);
+
+    assert.equal(modBold(1).color, 'red');  // Group value
+    assert(fontOption(1).multi);            // Group value
+    assert.equal(modBold(6).color, 'blue'); // node instance value
+    assert(fontOption(6).multi);            // Group value
+
+
+    network.setOptions({
+      groups: {
+        group1: {
+          //font: { color: 'brown' },  // Can not just change one field, entire font object is reset
+          font: {
+            multi: true,
+            color: 'brown'
+          },
+        },
+      },
+    });
+
+    assert.equal(modBold(1).color, 'brown'); // New value
+    assert(fontOption(1).multi);             // Group value
+    assert.equal(modBold(6).color, 'blue');  // unchanged
+    assert(fontOption(6).multi);             // unchanged
+
+
+    network.setOptions({
+      groups: {
+        group1: {
+          font: null,   // Remove font from group
+        },
+      },
+    });
+
+    // console.log("===============");
+    // console.log(fontOption(1));
+
+    assert.equal(modBold(1).color, '#343434');  // Reverts to default 
+    assert(!fontOption(1).multi);               // idem 
+    assert.equal(modBold(6).color, 'blue');     // unchanged
+    assert(fontOption(6).multi);                // unchanged
+
+    done();
+  });
+
```

Comment:
Great addition of tests! Well done!

I would suggest, however, refactoring these very large tests into a number of much smaller tests, in separate test classes where required, and making use of `before` and `after` to set up the state you need.

Reply:
I guess; they're kind of lengthy, that's true. I'll have a think about how to carve this up.

Reply:
Excellent! The only other suggestion I'll make is to follow the mocha convention for `pending`, or write the entire test and use `skip`, as described here: https://mochajs.org

This is a nit pick, and I'm approving as-is, with or without my suggestion.